app/models/notification.rb

```diff
@@ -38,4 +38,77 @@ def messagerie?
   def annotations_privees?
     Notification::ANNOTATIONS_PRIVEES.include?(type_notif)
   end
+
+  def self.for_dossier(gestionnaire, dossier)
+    follow = Follow
+      .includes(dossier: [:champs, :avis, :commentaires])
+      .find_by(gestionnaire: gestionnaire, dossier: dossier)
+
+    if follow.present?
+      #retirer le view_at.present? une fois la contrainte de presence en base (et les migrations ad hoc)
+      demande = follow.demande_view_at.present? &&
+        follow.dossier.champs.any? { |c| c.updated_at > follow.demande_view_at }
+
+      annotations_privees = follow.annotations_privees_view_at.present? &&
+        follow.dossier.champs_private.any? { |c| c.updated_at > follow.annotations_privees_view_at }
+
+      avis_notif = follow.avis_view_at.present? &&
+        follow.dossier.avis.any? { |avis| avis.updated_at > follow.avis_view_at }
+
+      messagerie = follow.messagerie_view_at.present? &&
+        dossier.commentaires.any? do |commentaire|
+        commentaire.updated_at > follow.messagerie_view_at &&
```

Comment:
Sauter une ligne avant pour améliorer la légibilité ?

Reply:
nop, j'ai rubocop qui me dit extra line at a begining of a block

Reply:
:(